### PR TITLE
Bring examples up to date with latest ReasonJS

### PR DESCRIPTION
**Pull request by sgrove**
Probably worth having @glennsl taken a look over it (especially the repeated `external domAsHtmlElement : ReasonJs.Dom.element => ReasonJs.Dom.htmlElement = "%identity";`), but this brings all the examples up to date with ReasonJs master

### CODE REVIEW

File: src/logo/logoRoot.re [hunk 1]
```diff
@@ -1 +1,4 @@
-ReactDOMRe.render <Logo message="REASON REACT" /> (ReasonJs.Document.getElementById "index");
+switch (ReasonJs.Document.getElementById "index" ReasonJs.Dom.document) {
+| None => Js.log "Root element 'index' not found on page, unable to start React app"
```
Comment (glennsl) [inline src/logo/logoRoot.re]: @chenglou will probably yell at you for not raising `Invalid_argument` here :)

File: src/simple/simpleRoot.re [hunk 2]
```diff
@@ -1,3 +1,6 @@
 Js.log "Do you see this message? If you do, then things should work!";
 
-ReactDOMRe.render <Page message="Hello!" /> (ReasonJs.Document.getElementById "index");
+switch (ReasonJs.Document.getElementById "index" ReasonJs.Dom.document) {
+| None => Js.log "Root element 'index' not found on page, unable to start React app"
```
Comment (glennsl) [inline src/simple/simpleRoot.re]: And here

File: src/todomvc/app.re [hunk 3]
```diff
@@ -1,3 +1,5 @@
+external domAsHtmlElement : ReasonJs.Dom.element => ReasonJs.Dom.htmlElement = "%identity";
```
Comment (glennsl) [inline src/todomvc/app.re]: You can use `Element.asHtmlElement` instead

File: src/todomvc/app.re [hunk 4]
```diff
@@ -171,4 +173,10 @@ module Top = {
   let createElement = wrapProps ();
 };
 
-ReactDOMRe.render <Top /> (ReasonJs.Document.getElementsByClassName "todoapp").(0);
+switch (
+  ReasonJs.HtmlCollection.item
+    0 (ReasonJs.Document.getElementsByClassName "todoapp" ReasonJs.Dom.document)
+) {
+| None => Js.log "Unable to find root 'todoapp' element, cannot start React app"
```
Comment (glennsl) [inline src/todomvc/app.re]: Invalid_argument here too probably

File: src/todomvc/todoItem.re [hunk 5]
```diff
@@ -1,3 +1,5 @@
+external domAsHtmlElement : ReasonJs.Dom.element => ReasonJs.Dom.htmlElement = "%identity";
```
Comment (glennsl) [inline src/todomvc/todoItem.re]: `Element.asHtmlElement` again

**Review by glennsl — COMMENTED** [review]
Looks pretty good! I did manage to find a few things though :)

**Comment by sgrove** [comment]
@glennsl Great suggestions, thanks!

Despite being part of the meta language, it'd be nice to have some community guide about when to raise like this, etc. I know I saw a comment from @chenglou on one of the issues about when to raise, but I don't recall where.

**Comment by chenglou** [comment]
Sorry, closing this as this has diverged and I've finished updating to the latest latest API. Thanks =)

**Pull request closed without merging**